components/raftstore-v2/src/operation/ready/apply_trace.rs

```diff
@@ -283,17 +312,40 @@ impl ApplyTrace {
                 }
             })
             .min();
+
         // At best effort, we can only advance the index to `mem_index`.
         let candidate = cmp::min(mem_index, min_flushed.unwrap_or(u64::MAX));
+        // always flush if there is any sst ingestion.
+        let has_ingested_sst = self.advance_flushed_index_for_ingest(candidate);
         if candidate > self.admin.flushed {
             self.admin.flushed = candidate;
-            if self.admin.flushed > self.persisted_applied + 100 {
+            if has_ingested_sst || (self.admin.flushed > self.persisted_applied + 100) {
                 self.try_persist = true;
             }
         }
         // TODO: persist admin.flushed every 10 minutes.
     }
 
+    fn advance_flushed_index_for_ingest(&mut self, max_index: u64) -> bool {
```

Comment:
I think what we want is use pending_sst_ranges to advance admin.flushed. The code here just decides whether to persist admin.flushed but not trully advance it.

Reply:
Actually, in the current logic sst ingestion doesn't not block the `admin.flushed` index advance because we don't make the applied_index as modified if it's an ingestion. So if the admin.flushed can't be advanced, it mean there is at least one cf that has not flushed write. 
The only change in this pr is to trigger global flushed index if it meets a ingestion, in the old logic, it will wait at least 100 index diff, so after this change we won't expect many ingestion reapply.

Reply:
For example, if we only have sst ingestion, say 10000 ingestions. Then our persisted_apply will not be changed and if restart, we will replay all of them.

Reply:
This is not correct. The bad case is that there are 10000 ingestion and a write after that, in the old logic, the index won't advance before the write is flushed. But anyway I have changed it as you said.